subcommand/acl-controller/command_test.go

```diff
@@ -421,3 +441,124 @@ func makeAuthMethod(principals interface{}) *api.ACLAuthMethod {
+		},
+		"mgw WAN fed enabled, primary DC, policy exists": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			existingPolicy: true,
+			expPolicy:      expOSSAnonTokenPolicy,
+		},
+		"mgw WAN fed enabled, primary DC, create policy OSS": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			expPolicy: expOSSAnonTokenPolicy,
+		},
+		"mgw WAN fed enabled, primary DC, create policy ENT": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			partitionsEnabled: true,
+			expPolicy:         expEntAnonTokenPolicy,
+		},
+	}
+	for name, c := range cases {
+		c := c
+		t.Run(name, func(t *testing.T) {
+			cfg := testutil.ConsulServer(t, testutil.ConsulACLConfigFn)
+			consulClient, err := api.NewClient(cfg)
+			require.NoError(t, err)
+
+			cmd := Command{
+				log:                   hclog.Default().Named("acl-controller"),
+				flagPartitionsEnabled: c.partitionsEnabled,
+			}
+
+			// if we're simulating that the policy already exists, then create it.
+			var expPolicyID string
+			if c.existingPolicy {
+				policy, _, err := consulClient.ACL().PolicyCreate(&api.ACLPolicy{
+					Name:        anonPolicyName,
+					Description: "Anonymous token policy",
+					Rules:       c.expPolicy,
+				}, &api.WriteOptions{})
+				require.NoError(t, err)
+				expPolicyID = policy.ID
+			}
+
+			expAnonToken, _, err := consulClient.ACL().TokenRead(anonTokenID, &api.QueryOptions{})
+			require.NoError(t, err)
+
+			err = cmd.upsertAnonymousTokenPolicy(consulClient, c.agentConfig)
+
+			if len(c.expErr) == 0 {
+				require.NoError(t, err)
+				obsAnonToken, _, err := consulClient.ACL().TokenRead(anonTokenID, &api.QueryOptions{})
+				require.NoError(t, err)
+
+				if len(c.expPolicy) > 0 {
+					// if we expect the policy to be created then read it back to make sure
+					// it was and that it matches the expected
+					obsAnonTokenPolicy, _, err := consulClient.ACL().PolicyReadByName(anonPolicyName, &api.QueryOptions{})
+					require.NoError(t, err)
+					require.Equal(t, c.expPolicy, obsAnonTokenPolicy.Rules)
+
+					if expPolicyID == "" {
+						// if the policy was created during the upsert then set the ID to the
+						// value that we got back when reading the policy.
+						expPolicyID = obsAnonTokenPolicy.ID
+					}
+
+					// expect that the policy is now attached to the anonymous token.
+					expAnonToken.Policies = append(expAnonToken.Policies, &api.ACLTokenPolicyLink{
+						ID:   expPolicyID,
+						Name: anonPolicyName})
+				}
+				// remove fields that don't compare
```

Comment:
Another way we've ignored fields in comparisons is with [`cmpopts.IgnoreFields`](https://github.com/hashicorp/consul-ecs/blob/b1bd110b128f62bd136a453c75a4faa1ef4df97f/subcommand/mesh-init/command_test.go#L293-L298)